pkg/agent/clientset.go

```diff
@@ -36,9 +36,12 @@ type ClientSet struct {
 	clients map[string]*Client // map between serverID and the client
 	// connects to this server.
 
-	agentID     string // ID of this agent
-	address     string // proxy server address. Assuming HA proxy server
-	serverCount int    // number of proxy server instances, should be 1
+	agentID string // ID of this agent
+	address string // proxy server address. Assuming HA proxy server
+
+	serverCounter           *ServerLeaseCounter // counts number of proxy server leases
```

Comment:
I think this is misleading. I don't believe this is actually a ServerLeaseCounter. It seems like a ServerCounterManager. One of the options in the ServerCounterManager is lease but thats just 1 of the options.

Reply:
Or maybe it should be misleading. I think it would be better to isolate the logic about server counts in 1 place rather than having various conditionals littering the server count throughout.

Reply:
Alternately the variable should be named leaseCounter if we don't want to restructure

Reply:
Changed to `leaseCounter`.